### ISSUE TITLE
Need test for #170

### ISSUE THREAD

**Issue by hiqsol** [yiisoft/di#171]
What exactly causes the problem?
Is it redifintion of `ContainerInterface` in providers?

**Comment by samdark** [comment]
Yes. That is changed order of execution. The fix doesn't apply to providers anymore. Likely it will be resolved by #143
